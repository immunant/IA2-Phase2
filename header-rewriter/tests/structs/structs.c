#include "structs.h"
#include <stdio.h>

struct s1 get_s1(void) {
	struct s1 s = {
		3,
		96,
	};
	return s;
}

int cksum_s1(struct s1 s) {
	return s.i1 + s.i2;
}

struct s2 get_s2(void) {
	struct s2 s = {
		3.14,
	};
	return s;
}

float extract_s2(struct s2 s) {
	return s.f1;
}

struct s3 get_s3(void) {
	struct s3 s = {
		.i1 = 50,
		.c1 = 65,
		.u1 = 65379,
	};
	return s;
}

int cksum_s3(struct s3 s) {
	return s.i1 + s.c1 + s.u1;
}

struct s3 inc_s3(struct s3 s) {
	s.i1++;
	s.c1++;
	s.u1++;
	return s;
}

struct s4 get_s4(void) {
	struct s4 s = {
		.f1 = 3.14,
		.f2 = 0.0016,
	};
	return s;
}

float cksum_s4(struct s4 s) {
	return s.f1 + s.f2;
}

struct s5 get_s5(void) {
	struct s5 s = {
		.i1 = 31,
		.p1 = (void*)0x4000030,
		.ac1 = {1, 2, 3},
		.p2 = (void*)0xffffbeef,
		.f1 = 98.7,
	};
	return s;
}

struct s5 get_s5_int(int i1) {
	struct s5 s = {
		.i1 = i1,
		.p1 = (void*)0x4000030,
		.ac1 = {1, 2, 3},
		.p2 = (void*)0xffffbeef,
		.f1 = 98.7,
	};
	return s;
}

void print_s5(struct s5 s) {
	printf("s5 { %d %p [%d,%d,%d] %p %.4f }\n",
		s.i1, s.p1, s.ac1[0], s.ac1[1], s.ac1[2], s.p2, s.f1);
}

float cksum_s5_f(struct s5 s) {
	return s.f1;
}
size_t cksum_s5_z(struct s5 s) {
	return s.i1 + (size_t)s.p1 + s.ac1[0] + s.ac1[1] + s.ac1[2] + (size_t)s.p2;
}

struct s6 get_s6(void) {
	struct s6 s = {
		.i1 = 8043,
		.c1 = 4,
		.i2 = 695465,
		.c2 = 100,
		.z1 = 0x9400000,
	};
	return s;
}

size_t cksum_s6(struct s6 s) {
	return s.i1 + s.c1 + s.i2 + s.c2 + s.z1;
}

struct s6 inc_s6(struct s6 s) {
	s.i1++;
	s.c1++;
	s.i2++;
	s.c2++;
	s.z1++;
	return s;
}

struct s6 mix_s6(struct s6 s1, struct s6 s2) {
	s1.i1 += s2.i1 / 2;
	s1.c1 += s2.c1 / 2;
	s1.i2 += s2.i2 / 2;
	s1.c2 += s2.c2 / 2;
	s1.z1 += s2.z1 / 2;
	return s1;
}

struct s7 get_s7(void) {
	struct s7 s = {
		.b1 = true,
		.c1 = 55,
		.u1 = 454000,
	};
	return s;
}

unsigned int cksum_s7(struct s7 s) {
	return s.b1 + s.c1 + s.u1;
}

struct s8 get_s8(void) {
	struct s8 s = {
		.b1 = true,
		.c1 = 15,
		.z1 = 454000,
		.c2 = 13,
		.z2 = 9685031,
		.z3 = 62121255,
	};
	return s;
}

size_t cksum_s8(struct s8 s) {
	return s.b1 + s.c1 + s.z1 + s.c2 + s.z2 + s.z3;
}

struct s9 get_s9(void) {
	struct s9 s = {
		.i1 = 96506328,
		.i2 = 1846,
		.i3 = 3254,
	};
	return s;
}

__int128 cksum_s9(struct s9 s) {
	return s.i1 + s.i2 + s.i3;
}

struct s10 get_s10(void) {
	struct s10 s = {
		.i1 = 96506328,
		.i2 = 1846,
		.i3 = 3254,
		.z1 = 6853785,
	};
	return s;
}

__int128 cksum_s10(struct s10 s) {
	return s.i1 + s.i2 + s.i3 + s.z1;
}

struct s11 get_s11(void) {
	struct s11 s = {
		.az1 = {1, 2, 3, 4, 5, 6, 7, 8, 9},
	};
	return s;
}

unsigned int cksum_s11(struct s11 s) {
	return 9 * s.az1[0]
		+ 8 * s.az1[1]
		+ 7 * s.az1[2]
		+ 6 * s.az1[3]
		+ 5 * s.az1[4]
		+ 4 * s.az1[5]
		+ 3 * s.az1[6]
		+ 2 * s.az1[7]
		+ 1 * s.az1[8];
}

struct s12 get_s12(void) {
	struct s12 s = {
		.ac1 = {1, 2, 3, 4, 5, 6, 7, 8, 9},
	};
	return s;
}

unsigned int cksum_s12(struct s12 s) {
	return 9 * s.ac1[0]
		+ 8 * s.ac1[1]
		+ 7 * s.ac1[2]
		+ 6 * s.ac1[3]
		+ 5 * s.ac1[4]
		+ 4 * s.ac1[5]
		+ 3 * s.ac1[6]
		+ 2 * s.ac1[7]
		+ 1 * s.ac1[8];
}

struct s13 get_s13(void) {
    struct s13 s = {
        .x = 0x7fffeeeeddddcccc
    };
    s.x <<= 64;
    s.x |= 0xbbbbaaaa99998888;
}

unsigned int cksum_s13(struct s13 s) {
    unsigned int x0 = s.x & 0xffffffff;
    unsigned int x1 = (s.x >> 32) & 0xffffffff;
    unsigned int x2 = (s.x >> 64) & 0xffffffff;
    unsigned int x3 = (s.x >> 96) & 0xffffffff;
    return x0 + x1 + x2 + x3;
}
