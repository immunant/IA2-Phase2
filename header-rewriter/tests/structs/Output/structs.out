cksum_s1(s) = 179 (expected 179)
s.i1 = 3 (expected 3)
s.i2 = 96 (expected 96)
extract_s2(s) = 999.9900 (expected 999.9900)
s.i1 = 50 (expected 50)
s.c1 = 65 (expected 65)
s.u1 = 65379 (expected 65379)
cksum_s4(s) = 52881.2500 (expected 52881.2500)
s.f1 = 3.1400 (expected 3.1400)
s.f2 = 0.0016 (expected 0.0016)
s5 { 6976 0xb17ebee7 [78,45,32] 0xba5edba5edba5ed 390.5600 }
cksum_s5_f(s) = 390.5600 (expected 390.5600)
cksum_s5_z(s) = 839338293263499439 (expected 839338293263499439)
s.i1 = 31 (expected 31)
s.p1 = 0x4000030 (expected 0x4000030)
s.ac1[0] = 1 (expected 1)
s.ac1[1] = 2 (expected 2)
s.ac1[2] = 3 (expected 3)
s.p2 = 0xffffbeef (expected 0xffffbeef)
s.f1 = 98.7000 (expected 98.7000)
s.i1 = 99954 (expected 99954)
cksum_s6(s) = 1000374 (expected 1000374)
s.i1 = 101 (expected 101)
cksum_s6(s) = 1000379 (expected 1000379)
s.i1 = 8043 (expected 8043)
s.c1 = 4 (expected 4)
s.i2 = 695465 (expected 695465)
s.c2 = 100 (expected 100)
s.z1 = 155189248 (expected 155189248)
s.i1 = 4122 (expected 4122)
s.c1 = 15 (expected 15)
s.i2 = 347989 (expected 347989)
s.c2 = 58 (expected 58)
s.z1 = 78594624 (expected 78594624)
cksum_s7(s) = 4034059 (expected 4034059)
s.b1 = 1 (expected 1)
s.c1 = 55 (expected 55)
s.u1 = 454000 (expected 454000)
cksum_s8(s) = 999357496 (expected 999357496)
s.b1 = 1 (expected 1)
s.c1 = 15 (expected 15)
s.z1 = 454000 (expected 454000)
s.c2 = 13 (expected 13)
s.z2 = 9685031 (expected 9685031)
s.z3 = 62121255 (expected 62121255)
cksum_s9(s) = 00000000000000000000000000000fd3 (expected 00000000000000000000000000000fd3)
s.i1 = 96506328 (expected 96506328)
s.i2 = 1846 (expected 1846)
s.i3 = 3254 (expected 3254)
cksum_s10(s) = 000000000000000000000000026b9a66 (expected 000000000000000000000000026b9a66)
s.i1 = 00000000000000000000000005c091d8 (expected 00000000000000000000000005c091d8)
s.i2 = 00000000000000000000000000000736 (expected 00000000000000000000000000000736)
s.i3 = 00000000000000000000000000000cb6 (expected 00000000000000000000000000000cb6)
s.z1 = 6853785 (expected 6853785)
cksum_s11(s) = 2538 (expected 2538)
s.az1[0] = 1 (expected 1)
s.az1[1] = 2 (expected 2)
s.az1[2] = 3 (expected 3)
s.az1[3] = 4 (expected 4)
s.az1[4] = 5 (expected 5)
s.az1[5] = 6 (expected 6)
s.az1[6] = 7 (expected 7)
s.az1[7] = 8 (expected 8)
s.az1[8] = 9 (expected 9)
cksum_s12(s) = 2538 (expected 2538)
s.ac1[0] = 1 (expected 1)
s.ac1[1] = 2 (expected 2)
s.ac1[2] = 3 (expected 3)
s.ac1[3] = 4 (expected 4)
s.ac1[4] = 5 (expected 5)
s.ac1[5] = 6 (expected 6)
s.ac1[6] = 7 (expected 7)
s.ac1[7] = 8 (expected 8)
s.ac1[8] = 9 (expected 9)
cksum_s14(s) = 2 (expected 2)
s.u1 = 0 (expected 0)
s.u2 = 1 (expected 1)
s.u3 = 0 (expected 0)
s.u4 = 0 (expected 0)
s.u5 = 1 (expected 1)
s.u6 = 0 (expected 0)
s.u7 = 0 (expected 0)
s.u8 = 0 (expected 0)
s.u9 = 1 (expected 1)
s.u10 = 0 (expected 0)
s.u11 = 0 (expected 0)
s.u12 = 0 (expected 0)
s.u13 = 0 (expected 0)
s.u14 = 1 (expected 1)
s.u15 = 0 (expected 0)
s.u16 = 0 (expected 0)
s.u17 = 0 (expected 0)
s.u18 = 0 (expected 0)
s.u19 = 0 (expected 0)
s.u20 = 1 (expected 1)
s.u21 = 0 (expected 0)
s.u22 = 0 (expected 0)
s.u23 = 0 (expected 0)
s.u24 = 0 (expected 0)
s.u25 = 0 (expected 0)
s.u26 = 1 (expected 1)
s.u27 = 0 (expected 0)
s.u28 = 1 (expected 1)
s.u29 = 1 (expected 1)
s.u30 = 1 (expected 1)
s.u31 = 1 (expected 1)
s.u32 = 0 (expected 0)
s.field2 = 20000 (expected 20000)
s.field3 = 30000 (expected 30000)
s.field4 = 40000 (expected 40000)
s.field5 = 50000 (expected 50000)
s.field6 = 60000 (expected 60000)
s.field7 = 70000 (expected 70000)
s.field8 = 80000 (expected 80000)
